content/blog/2021-02-17-cml-runner-prerelease.md

```diff
@@ -0,0 +1,165 @@
+workflow and then train a model.
+
+```yaml
+name: 'Train in the cloud'
+on: [push]
+
+jobs:
+  deploy-runner:
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: iterative/setup-cml@v1
+      - uses: actions/checkout@v2
+      - name: deploy
+        shell: bash
+        env:
+          repo_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cml-runner \
+          --cloud aws \
+          --cloud-region us-west \
+          --cloud-type=t2.micro \
+          --labels=cml-runner
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted, cml-runner]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: 'Train my model'
+        run: |
+          pip install -r requirements.txt
+          python train.py
+```
+
+If you use CML functions in the `train-model` step, you can go even further and
+get a closed loop- sending model training results from the EC2 instance to your
+pull request or merge request! For example, if we expand the `train-model` step
+to incorporate functions like `cml-publish` and `cml-send-comment`:
+
+```yaml
+train-model:
+  needs: deploy-runner
+  runs-on: [self-hosted, cml-runner]
+  container: docker://dvcorg/cml-py3
+  steps:
+    - uses: actions/checkout@v2
+    - name: 'Train a model'
+      env:
+        repo_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+      run: |
+        pip install -r requirements.txt
+        python train.py
+
+        echo "## Report from your EC2 Instance" > report.md
+        cat metrics.txt >> report.md
+        cml-publish "confusion_matrix.png" --md >> report.md
+        cml-send-comment report.md
+```
+
+You'll get a pull request that looks something like this:
+
+![](/uploads/images/2021-02-17/sample_pr.png)
+
+All the code to replicate this example is up on a
+[brand new demo repository](https://github.com/iterative/cml-runner-base-case).
+
+### Our favorite details
+
+The new `cml-runner` function lets you turn on instances, including GPU,
+high-memory and sport instances, and kick off a new workflow using the hardware
+and environment of your choice- and of course, it'll turn _off_ those instances
+after a configurable timeout!
+
+Another highlight: you can use whatever Docker container you'd like on your
+instance. In the above example, we use our
+[custom CML Docker container](https://github.com/iterative/cml/blob/master/docker/Dockerfile)
+(because we like it!) - but you certainly don't have to! Whatever image you
+choose, we highly recommend containerizing your environment for ultimate
+reproducibility and security with CML.
+
+You can also use the new `cml-runner` function to set up a local self-hosted
+runner. On the local machine, you'll install CML as a package and then run:
```

Comment:
self-hosted runner - a link to GH/GL docs might be helpful.

`On the local machine` - I'm not sure it is a primary use-case for self hosted runners.
It can be `On the local machine or your on-premise GPU cluster`